### PR TITLE
check ad showing state to avoid false positives

### PR DESCRIPTION
**Pull request by jdorweiler**
**Asana Task/Github Issue:** 
https://app.asana.com/1/137249556945/project/11472677167855/task/1215688422963506

## Description

Limit false positives in static ad detection. 

## Testing Steps

- 

## Checklist

*Please tick all that apply:*

- [ ] I have tested this change locally
- [ ] I have tested this change locally in all supported browsers
- [ ] This change will be visible to users
- [ ] I have added automated tests that cover this change
- [ ] I have ensured the change is gated by config
- [ ] This change was covered by a ship review
- [ ] This change was covered by a tech design
- [ ] Any dependent config has been merged

---

> [!NOTE]
> **Low Risk**
> Narrow change to static-ad heuristics in the injected detector with new ab tests; may miss edge cases where static ads lack those player classes.
> 
> **Overview**
> **Static overlay ad detection** no longer fires on the normal pre-play/cued poster. The same background DOM used for real static ads also appears before playback; this change requires the player root to carry **`ad-showing`** or **`ad-interrupting`** before static selectors are evaluated.
> 
> The detection sweep now passes its resolved **`root`** into **`checkForStaticAds`**, with fallback to cached or discovered player root. Unit tests cover poster (no ad state), **`ad-showing`**, and **`ad-interrupting`**.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 48b7cde2dc2adf9e67512fe35de91bca9a53f71f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: injected/src/detectors/detections/youtube-ad-detection.js [hunk 1]
```diff
@@ -318,6 +318,17 @@ export class YouTubeAdDetector {
...
             return false;
         }
+
+        // A genuine static ad renders while the player is in its ad state
+        // (#movie_player.ad-showing / .ad-interrupting). The creator pre-play/cued
+        // poster uses the same `.player-container-background` container WITHOUT the ad
+        // state, so without this gate it is mis-detected as a static ad (false positive).
+        // Reuse the config-driven player root rather than a separate hardcoded query.
+        const player = this.playerRoot ?? this.findPlayerRoot();
+        if (!player || !/\bad-showing\b|\bad-interrupting\b/.test((player.className || '').toString())) {
```
[the remark this answers is not in the document]
Comment (jdorweiler) [reply injected/src/detectors/detections/youtube-ad-detection.js:329]: could cause false positives

File: injected/src/detectors/detections/youtube-ad-detection.js [hunk 2]
```diff
@@ -311,13 +311,25 @@ export class YouTubeAdDetector {
...
             return false;
         }
+
+        // A genuine static ad renders while the player is in its ad state
+        // (#movie_player.ad-showing / .ad-interrupting). The creator pre-play/cued
+        // poster uses the same `.player-container-background` container WITHOUT the ad
+        // state, so without this gate it is mis-detected as a static ad (false positive).
+        // Reuse the sweep's already-resolved (config-driven) player root when available.
+        const player = root ?? this.playerRoot ?? this.findPlayerRoot();
+        if (!player || !/\bad-showing\b|\bad-interrupting\b/.test((player.className || '').toString())) {
```
[the remark this answers is not in the document]
Comment (jdorweiler) [reply injected/src/detectors/detections/youtube-ad-detection.js:329]: do this later. I don't want to complicated the other in progress event hub implementations

**Review by daxtheduck — DISMISSED** [review]

**Review by GuiltyDolphin — APPROVED** [review]

**Review by daxtheduck — DISMISSED** [review]

**Review by daxtheduck — DISMISSED** [review]

**Review by daxtheduck — APPROVED** [review]
